pkg/config/config.go

```diff
@@ -155,18 +226,18 @@ func (lci *LocalConfigInfo) GetConfiguration(parameter string) (interface{}, boo
 
 	switch strings.ToLower(parameter) {
```

Comment:
Can we remove `GetConfiguration` function? It is returning `interface{}` :-( We have getters for every field

Reply:
need `GetConfiguration` for `odo config view`

Reply:
Now we are handling default values in two places :-( It will lead to inconsistencies. 

Getters for each field should be the only source of truth for the config values.

Reply:
You are using `GetComponentSettings` not `GetConfiguration` in `view.go`

Reply:
I am using `GetConfiguration` in `set` and `unset` and I don't think we can replace that.

Reply:
You are using it just to check if the config is set or not, but not for anything else. You can create `IsSet` function for that that will be much simpler.
This will have a huge maintenance cost, to manage two different ways to read the config and making sure that you handle default values etc.. in the same way.

Reply:
I get it but its not just about checking if the value is set, if the value is set, we show the user
`the value is already set. The current value being %v`. 
I will not be able to show the current value if we change this to `IsSet`. We can consider a reflection mechanism in the future where the `GetConfiguration` instead of directly accessing the attributes uses the getters ( via reflection ). This way the maintain ability stays.

Reply:
To be honest, I would rather stop showing what the current value of the config is than having to manage the same logic in two places.